### PR TITLE
8284752: Zero does not build on Mac OS X due to missing os::current_thread_enable_wx implementation

### PR DESCRIPTION
**Pull request by parttimenerd**
Adds an implementation of the missing method (guarded with `defined(AARCH64) && defined(__APPLE__)`) and fixes the compilation issues on Mac M1.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8284752](https://bugs.openjdk.java.net/browse/JDK-8284752): Zero does not build on Mac OS X due to missing os::current_thread_enable_wx implementation

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8200/head:pull/8200` \
`$ git checkout pull/8200`

Update a local copy of the PR: \
`$ git checkout pull/8200` \
`$ git pull https://git.openjdk.java.net/jdk pull/8200/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8200`

View PR using the GUI difftool: \
`$ git pr show -t 8200`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8200.diff">https://git.openjdk.java.net/jdk/pull/8200.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/os_cpu/bsd_zero/os_bsd_zero.cpp [hunk 1]
```diff
@@ -350,3 +350,9 @@ int os::extra_bang_size_in_bytes() {
   // Zero does not require an additional stack bang.
   return 0;
 }
+
+#if defined(AARCH64) && defined(__APPLE__)
+void os::current_thread_enable_wx(WXMode mode) {
+  pthread_jit_write_protect_np(mode == WXExec);
+}
+#endif
```
Comment (dholmes-ora) [inline src/hotspot/os_cpu/bsd_zero/os_bsd_zero.cpp:358]: Newline needed at end-of-file

**Review by dholmes-ora — APPROVED** [review]
Looks good and trivial.

It seems apparent that nobody has been building Zero in this environment.

Thanks,
David

**Comment by parttimenerd** [comment]
Yes, it does. But I included it in my [jdk-profiling-tester](https://github.com/parttimenerd/jdk-profiling-tester) to find crashes causes by AsyncGetCallTrace.

**Review by dholmes-ora — APPROVED** [review]

**Comment by shipilev** [comment]
> I'm a bit surprised at this. I thought macos/aarch64 was added to GHA some time ago? (And also to Oracle's internal CI pipeline)

This is Zero-specific, note the compilation unit name :)

**Pull request closed without merging**